CMakeLists.txt

```diff
@@ -1,20 +1,130 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.12)
 
-project(ozo)
+project(ozo VERSION 0.0.1)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+
+include(CheckCXXCompilerFlag)
+
+function (target_supported_compile_options target)
+    set(options BEFORE)
+    set(arguments INTERFACE PUBLIC PRIVATE)
+    cmake_parse_arguments(OPTION "${options}" "" "${arguments}" ${ARGN})
+
+    foreach(compile_option ${OPTION_INTERFACE})
```

Comment:
I think this code may be simplified if variable name would be generated like [this](https://stackoverflow.com/questions/32631168/cmake-interpret-string-as-variable)

Reply:
I don't really understand your point here, but I have removed the code now, in favor of hard-coding the flags.

Reply:
The point is to do not have three almost identical `foreach` statement, instead of one which could be parametrized by the option's scope set.